lib/active_admin/middleware/error_handler.rb

```diff
@@ -0,0 +1,26 @@
+module ActiveAdmin
+  class ErrorHandler
+
+    def initialize(app)
+      @app = app
+    end
+
+    def call(env)
+      @app.call(env)
+    rescue Exception => e
+      if ActiveAdmin.application.default_namespace.to_s == env['PATH_INFO'].split('/').second
+        render_exception(env, e)
+      else
+        @app.call(env)
+      end
+    end
```

Comment:
a cleaner solution:
```ruby
      def call(env)
        @app.call(env)
      rescue Exception => exception
        if is_active_admin_error?
          render_exception(env, exception)
        else
          raise exception
        end
      end

      private

      def is_active_admin_error?
        ActiveAdmin.application.default_namespace.to_s == env['PATH_INFO'].split('/').second
      end
```
In your version you double call the application in case of a error. Thats not good for two reasons, first it coats time, second after the first call you have a different situation that maybe raise a different error.